app/services/transaction_service/store/transaction.rb

```diff
@@ -0,0 +1,130 @@
+    [:automatic_confirmation_after_days, :fixnum, :mandatory],
+    [:minimum_commission, :money, :mandatory],
+    [:content, :string],
+    [:booking_fields, :hash])
+
+  Transaction = EntityUtils.define_builder(
+    [:id, :fixnum, :mandatory],
+    [:community_id, :fixnum, :mandatory],
+    [:listing_id, :fixnum, :mandatory],
+    [:starter_id, :string, :mandatory],
+    [:listing_quantity, :fixnum, :mandatory],
+    [:listing_title, :string, :mandatory],
+    [:listing_author_id, :string, :mandatory],
+    [:unit_price, :money, :mandatory],
+    [:payment_process, :to_symbol, one_of: [:none, :postpay, :preauthorize]],
+    [:payment_gateway, :to_symbol, one_of: [:paypal, :checkout, :braintree, :none]],
+    [:commission_from_seller, :fixnum, :mandatory],
+    [:automatic_confirmation_after_days, :fixnum, :mandatory],
+    [:minimum_commission, :money, :mandatory],
+    [:last_transition_at, :time],
+    [:current_state, :to_symbol])
+
+  FINISHED_TX_STATES = "'free', 'rejected', 'confirmed', 'canceled', 'errored'"
+
+  module_function
+
+  def create(opts)
+    tx_data = HashUtils.compact(NewTransaction.call(opts))
+    tx_model = TransactionModel.new(tx_data.except(:content, :booking_fields))
+    build_conversation(tx_model, tx_data)
+    build_booking(tx_model, tx_data)
+
+    tx_model.save!
+    from_model(tx_model)
+  end
+
+  def get(transaction_id)
+    Maybe(TransactionModel.where(id: transaction_id).first)
+      .map { |m| from_model(m) }
+      .or_else(nil)
+  end
+
+  def get_in_community(community_id:, transaction_id:)
+    Maybe(TransactionModel.where(id: transaction_id, community_id: community_id).first)
+      .map { |m| from_model(m) }
+      .or_else(nil)
+  end
+
+  def unfinished_tx_count(person_id)
+    TransactionModel
+      .where("starter_id = ? OR listing_author_id = ?", person_id, person_id)
+      .where("current_state NOT IN (#{FINISHED_TX_STATES})")
+      .count
+  end
+
+
+  ## Privates
+
+  def from_model(model)
+    Maybe(model)
+      .map { |m|
+        EntityUtils.model_to_hash(m)
+        .merge({unit_price: m.unit_price , minimum_commission: m.minimum_commission })
+      }
+      .map { |hash| Transaction.call(hash) }
+      .or_else(nil)
+  end
+
+  def build_conversation(tx_model, tx_data)
+    conversation = tx_model.build_conversation(
+      tx_data.slice(:community_id, :listing_id))
+
+    conversation.participations.build(
+      person_id: tx_data[:listing_author_id],
+      is_starter: false,
+      is_read: false)
+
+    conversation.participations.build(
+      person_id: tx_data[:starter_id],
+      is_starter: true,
+      is_read: true)
+
+    if tx_data[:content].present?
+      conversation.messages.build({
+          content: tx_data[:content],
+          sender_id: tx_data[:starter_id]})
+    end
+  end
+
+  def build_booking(tx_model, tx_data)
+    if is_booking?(tx_data)
+
+      # TODO What's the correct place for the booking calculation logic?
```

Comment:
Even more relevant with upcoming booking options. At the moment this just keeps the existing functionality which is to verify that the incoming listing_quantity matches with the booking dates. Let's revisit this when are further along with shipping / booking and see if this check should just be eliminated here.